reascripts/common/libs/PathUtil.lua

```diff
@@ -0,0 +1,91 @@
+--[[
+
+  PathUtil.lua - Path utilities for ReaScript
+
+]]--
+
+PathUtil = {}
+
+-- Apply an extension to a file path, if it doesn't already have one.
+-- Empty or nil path returns the empty string.
+PathUtil.apply_extension = function(filepath, extension)
+  if not filepath or #filepath < 1 then
+    return ""
+  end
+
+  local path = ""
+
+  local path_part, file_part = filepath:match("(.+)[\\/](.+)")
+
+  if path_part then
+    path = path_part .. PathUtil._path_separator()
+    filepath = file_part
+  end
+
+  -- split the path by ".", if last chunk is alphanumeric don't apply extension
+  local chunks = {}
+  for chunk in filepath:gmatch("[^%.]+") do
+    table.insert(chunks, chunk)
+  end
+
+  local last_chunk = chunks[#chunks]
+
+  if #chunks > 1 and last_chunk:find("%w+") then
+    return path .. filepath
+  end
+
+  return path .. filepath .. '.' .. extension
+end
+
+-- Returns the given path if a full path, otherwise returns given path
+-- relative to the REAPER project resource directory.
+PathUtil.get_real_path = function(path_arg)
+  if PathUtil._is_full_path(path_arg) then
+    return path_arg
+  end
+
+  local project_path = reaper.GetProjectPath()
+  return project_path .. PathUtil._path_separator() .. path_arg
+end
+
+-- Returns the command to reveal the given path in the OS file explorer.
+-- Works for Windows and macOS. To quote a wise philosopher, "Linux is a mess."
+PathUtil.get_reveal_command = function(path_arg)
+  if PathUtil._is_windows() then
+    return 'explorer /select,"' .. path_arg .. '"'
+  elseif PathUtil._is_mac() then
+    return 'open -R "' .. path_arg .. '"'
+  end
+
+  return ""
+end
+
+PathUtil._is_full_path = function(path)
+  local found
+
+  if PathUtil._is_windows() then
+    found = path:find("^%w:\\")
```

Comment:
Maybe we should also check for network (UNC) paths, e.g. `\\Server\Volume\File`

Reply:
totally forgot about those!